lib/OrderedMap.fram

```diff
@@ -0,0 +1,315 @@
+let false = False
+let true = True
+
+rec 
+  data Color = Red | Black
+  data Tree Value = Leaf | Node of Color, Tree Value , Value , Tree Value
+end
+
+method empty {X,self : Tree X} = Leaf
+
+let isEmpty tree = 
+  match tree with
+  | Leaf => true
+  | _ => false
+  end
+
+method isEmpty {X, self : Tree X} = isEmpty self
+
+let id x = x
+
+let balance tree =
+  match tree with
+  | Node Black (Node Red (Node Red a x b) y c) z d =>
+    Node Red (Node Black a x b) y (Node Black c z d)
+  | Node Black (Node Red a x (Node Red b y c)) z d =>
+    Node Red (Node Black a x b) y (Node Black c z d)
+  | Node Black a x (Node Red (Node Red b y c) z d) =>
+    Node Red (Node Black a x b) y (Node Black c z d)
+  | Node Black a x (Node Red b y (Node Red c z d)) =>
+    Node Red (Node Black a x b) y (Node Black c z d)
+  | x => x
+end 
+
+let paintItBlack tree = 
+  match tree with
+  | Node Red a x b =>  Node Black a x b
+  | _ => tree
+  end
+
+let repaint tree = 
+  match tree with
+  | Node Red _ _ _ => (False,paintItBlack tree)
+  | _ => (True, tree)
+  end
+
+let chkDelL color tree = 
```

Comment:
Please use more descriptive function names, someone will come after you and have maintain it.